cli-plugins/manager/manager.go

```diff
@@ -145,16 +138,20 @@ func getPlugin(name string, pluginDirs []string, rootcmd *cobra.Command) (*Plugi
 
 // ListPlugins produces a list of the plugins available on the system
 func ListPlugins(dockerCli config.Provider, rootcmd *cobra.Command) ([]Plugin, error) {
-	pluginDirs, err := getPluginDirs(dockerCli.ConfigFile())
-	if err != nil {
-		return nil, err
-	}
-
+	pluginDirs := getPluginDirs(dockerCli.ConfigFile())
 	candidates := listPluginCandidates(pluginDirs)
+	if len(candidates) == 0 {
+		return nil, nil
+	}
 
 	var plugins []Plugin
 	var mu sync.Mutex
-	eg, _ := errgroup.WithContext(context.TODO())
+	ctx := rootcmd.Context()
+	if ctx == nil {
+		// Fallback, mostly for tests that pass a bare cobra.command
+		ctx = context.Background()
+	}
+	eg, _ := errgroup.WithContext(ctx)
```

Comment:
maybe we should instead force the tests to set a context?

Reply:
Yeah, I originally updated the test, but given that this function is exported, I could not 100% exclude the possibility for other code to call this with a command that has not (yet?) has a context set.

Without a context, this produces a panic at runtime, which was a bit risky, so ultimately I chose for making this "best effort", and keep the test as-is to verify that things work even if no context is present on the command.

Reply:
Ah i see, alright then let's get this one merged.